src/matlab2tikz.m

```diff
@@ -3569,8 +3561,20 @@ function warnFacetedInterp(m2t)
                        env, drawOpts, tabOpts, table);
 end
 % ==============================================================================
+function marker = getMarkerInfo(m2t, h)
+    % gets marker-related options as a struct
+    marker                        = struct();
+    marker.style                  = get(h, 'Marker');
+    marker.FaceColor              = get(h, 'MarkerFaceColor');
+    marker.EdgeColor              = get(h, 'MarkerEdgeColor');
+    marker.hasFaceColor           = ~isNone(marker.FaceColor);
+    marker.hasEdgeColor           = ~isNone(marker.EdgeColor);
+    [marker.tikz, marker.options] = translateMarker(m2t, marker.style, ...
+        opts_new(), marker.hasFaceColor);
```

Comment:
Yeah, but generally we have the pattern 
```
[m2t, *Opt] = Func(m2t, h, *Opt, Arg)
```
Which we should stick to, or as you said put it inside the function as below.
```
[m2t, *Opt] = Func(m2t, h, Arg)
```